compose.go

```diff
@@ -0,0 +1,86 @@
+package sqlhooks
+
+import (
+	"context"
+	"fmt"
+)
+
+// Compose allows for composing multiple Hooks into one.
+// It runs every callback on every hook in argument order,
+// even if previous hooks return an error.
```

Comment:
should it work this way? A less surprising implementation would be to return on the first error. What is a use case of always running all hooks?

Reply:
@keegancsmith I can imagine the case when you:
`Compose(OpenTracing, Logs)`

OpenTracing might fail but you still want to execute Log

Reply:
pretty much what @gchaincl said. Another kind of  instrumentation you'd always want to happen might include recording of metrics.

I actually implemented the "bail on first error" strategy first, but decided to go with this approach instead. Happy to change it back though. WDYT?

Reply:
@keegancsmith ^ ?

Reply:
I suppose if OpenTracing or Logs fails, do we want to fail the SQL command? Personally my preference falls towards failing on the first error, even with the above mentioned use case. This is because that is more how I expect wrapping/function composition to work. Additionally it is more "simple": no error wrapping or expanding the public API required.

However, I am torn. I suspect the main use case for composing is multiple things which export telemetry, so this way works. So it is likely worth it, but maybe should be better documented. Additionally, can we avoid exporting the error type? Or do you think that is a requirement?